bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/CallOptionsFactory.java

```diff
@@ -98,6 +124,34 @@ public ConfiguredCallOptionsFactory(CallOptionsConfig config) {
       }
     }
 
+    @Override
+    public <RequestT> CallOptions createForRpcAttempt(RequestT request, Deadline currentDeadline) {
+      if (config.isUseTimeout() && request != null) {
+        long timeoutMs = getRequestAttemptTimeout(request);
+        long chosenTimeoutMs;
+        if (currentDeadline != null) {
+          // Ensure we don't exceed the current absolute deadline.
+          chosenTimeoutMs = Math.min(timeoutMs, currentDeadline.timeRemaining(TimeUnit.MILLISECONDS));
+        } else {
+          chosenTimeoutMs = timeoutMs;
+        }
+        return CallOptions.DEFAULT.withDeadline(Deadline.after(chosenTimeoutMs, TimeUnit.MILLISECONDS));
+      } else {
+        return CallOptions.DEFAULT;
+      }
+    }
+
+    @Override
```

Comment:
If you use the logic above, you no longer need to check this as it will fallback correctly